gnomad_qc/v3/variant_qc/random_forest.py

```diff
@@ -0,0 +1,475 @@
+            hl.null(hl.tfloat32),
+            inbreeding_ht.InbreedingCoeff,
+        )
+    )
+    trio_stats_ht = fam_stats.ht()
+    trio_stats_ht = trio_stats_ht.select(
+        f"n_transmitted_{group}", f"ac_children_{group}"
+    )
+
+    truth_data_ht = get_truth_ht()
+    allele_data_ht = allele_data.ht()
+    allele_counts_ht = qc_ac.ht()
+
+    logger.info("Annotating Table with all columns from multiple annotation Tables")
+    ht = ht.annotate(
+        **inbreeding_ht[ht.key],
+        **trio_stats_ht[ht.key],
+        **truth_data_ht[ht.key],
+        **allele_data_ht[ht.key].allele_data,
+        **allele_counts_ht[ht.key],
+    )
+    # Filter to only variants found in high quality samples and are not lowqual
+    ht = ht.filter(
+        (ht[f"ac_qc_samples_{group}"] > 0) & ~ht.AS_lowqual
+    )
+    ht = ht.select(
+        "a_index",
+        "was_split",
+        *FEATURES,
+        *TRUTH_DATA,
+        **{
+            "transmitted_singleton": (ht[f"n_transmitted_{group}"] == 1)
+            & (ht[f"ac_qc_samples_{group}"] == 2),
+            "fail_hard_filters": (ht.QD < 2) | (ht.FS > 60) | (ht.MQ < 30),
+        },
+        singleton=ht.ac_release_samples_raw == 1,
+        ac_raw=ht.ac_qc_samples_raw,
+        ac=ht.ac_release_samples_adj,
+        ac_qc_samples_unrelated_raw=ht.ac_qc_samples_unrelated_raw
+    )
+
+    ht = ht.repartition(n_partitions, shuffle=False)
+    if checkpoint_path:
+        ht = ht.checkpoint(checkpoint_path, overwrite=True)
+
+    if impute_features_by_variant_type:
+        ht = median_impute_features(ht, {"variant_type": ht.variant_type})
+
+    summary = ht.group_by("omni", "mills", "transmitted_singleton",).aggregate(
+        n=hl.agg.count()
+    )
+    logger.info("Summary of truth data annotations:")
+    summary.show(20)
+
+    return ht
+
+
+def train_rf(
+    ht: hl.Table,
+    fp_to_tp: float = 1.0,
+    num_trees: int = 500,
+    max_depth: int = 5,
+    no_transmitted_singletons: bool = False,
+    no_inbreeding_coeff: bool = False,
+    vqsr_training: bool = False,
+    vqsr_type: str = False,
+    filter_centromere_telomere: bool = False,
+    test_intervals: Union[str, List[str]] = "chr20",
+):
+    """
+    Train random forest model using `train_rf_model`
+
+    :param ht: Table containing annotations needed for RF training, built with `create_rf_ht`
+    :param fp_to_tp: Ratio of FPs to TPs for creating the RF model. If set to 0, all training examples are used.
+    :param num_trees: Number of trees in the RF model.
+    :param max_depth: Maxmimum tree depth in the RF model.
+    :param no_transmitted_singletons: Do not use transmitted singletons for training.
+    :param no_inbreeding_coeff: Do not use inbreeding coefficient as a feature for training.
+    :param vqsr_training: Use VQSR training sites to train the RF.
+    :param vqsr_type: VQSR model to use for vqsr_training. `vqsr_training` must be True for this parameter to be used.
```

Comment:
Yeah this is probably confusing, but flavor is the word that Laurent was using, but really we just want the VQSR model ID because they really mean the same thing, this it to grab the correct model that was run using `get_vqsr_filters` I will call it `vqsr_model_id`